pilot/pkg/serviceregistry/kube/controller/authorization.go

```diff
@@ -38,30 +40,264 @@ func (c *Controller) Policies(requested sets.Set[model.ConfigKey]) []*security.A
+		if k.Kind == kind.PeerAuthentication {
+			// PeerAuthentications are synthetic so prepend our special prefix
+			k.Name = fmt.Sprintf("%s-%s", convertedPeerAuthenticationPrefix, k.Name)
+		}
 		if len(requested) > 0 && !requested.Contains(k) {
 			continue
 		}
-		pol := convertAuthorizationPolicy(c.meshWatcher.Mesh().GetRootNamespace(), cfg)
-		if pol == nil {
+
+		var pol *security.Authorization
+		switch cfg.GroupVersionKind {
+		case gvk.AuthorizationPolicy:
+			pol = convertAuthorizationPolicy(c.meshWatcher.Mesh().GetRootNamespace(), cfg)
+			if pol == nil {
+				continue
+			}
+			res = append(res, pol)
+		case gvk.PeerAuthentication:
+			pol = convertPeerAuthentication(c.meshWatcher.Mesh().GetRootNamespace(), cfg)
+			if pol == nil {
+				continue
+			}
+			res = append(res, pol)
+		default:
+			log.Errorf("unknown config type %v", cfg.GroupVersionKind)
 			continue
 		}
-		res = append(res, pol)
 	}
+
+	// If there are any PeerAuthentications in our cache, send our static STRICT policy
+	if len(peerAuthenticationPolicies) > 0 {
+		res = append(res, &security.Authorization{
+			Name:      staticStrictPolicyName,
+			Namespace: c.meshWatcher.Mesh().GetRootNamespace(),
+			Scope:     security.Scope_WORKLOAD_SELECTOR,
+			Action:    security.Action_DENY,
+			Groups: []*security.Group{
+				{
+					Rules: []*security.Rules{
+						{
+							Matches: []*security.Match{
+								{
+									NotPrincipals: []*security.StringMatch{
+										{
+											MatchType: &security.StringMatch_Presence{},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		})
+	}
+
 	return res
 }
 
+// convertedSelectorPeerAuthentications returns a list of keys corresponding to one or both of:
+// [static STRICT policy, port-level STRICT policy] based on the effective PeerAuthentication policy
+func (c *Controller) convertedSelectorPeerAuthentications(ns string, lbls map[string]string) []string {
+	var meshCfg, namespaceCfg, workloadCfg *config.Config
+
+	// Whether it comes from a mesh-wide, namespace-wide, or workload-specific policy
+	// if the effective policy is STRICT, then reference our static STRICT policy
+	var isEffectiveStrictPolicy bool
+	// Only 1 per port workload policy can be effective at a time. In the case of a conflict
+	// the oldest policy wins.
+	var effectivePortLevelPolicyKey string
+
+	rootNamespace := c.meshWatcher.Mesh().GetRootNamespace()
+
+	matches := func(c config.Config) bool {
+		sel := c.Spec.(*v1beta1.PeerAuthentication).Selector
+		if sel == nil {
+			return false
+		}
+		return labels.Instance(sel.MatchLabels).SubsetOf(lbls)
+	}
+
+	configs := c.configController.List(gvk.PeerAuthentication, rootNamespace)
+	configs = append(configs, c.configController.List(gvk.PeerAuthentication, ns)...)
+
+	for i := range configs {
+		cfg := configs[i]
+		spec, ok := cfg.Spec.(*v1beta1.PeerAuthentication)
+
+		if !ok || spec == nil {
+			continue
+		}
+
+		// We don't support DISABLE policies in Ambient mode
```

Comment:
🤔 Hmm in my mind DISABLE would mean "block traffic if it IS tunneled/TLS" which is distinct from PERMISSIVE which allows both. The original [Ambient APIs doc](https://docs.google.com/document/d/1Lk1I1qB-XwVWFP0sVcIN5JMJZHjuF3j8lGWtUeDK6pY/edit#heading=h.xw1gqgyqs5b) also states that we'd disallow DISABLED PeerAuthn for ambient.

Reply:
But the intent isn't really to block TLS, IMO. I strongly suspect people are using DISABLE to mean "I do not always have TLS for <reasons>, and PERMISSIVE breaks me because it was poorly implemented". 

FWIW I wrote the original doc that said disallow and I hadn't considered merging or per-port, so 'disallow' was the same as 'treat as permissive' in my intent

Reply:
> I do not always have TLS for , and PERMISSIVE breaks me because it was poorly implemented

Why would PERMISSIVE break people? Because it will always requires TLS if a sidecar was injected (regardless of whether or not the connection was TLS)?

Reply:
Sniffing TLS is not reliable in general. Server-first protocols break it, or any protocol that treats connections as special is affected (NTLM?).

Reply:
ztunnel 'permissive' is fundamentally different since there is no sniffing, there is a dedicated tunnel port.

Basically like if you setup wireguard. You can chose to make that the _only_ interface (STRICT) or leave the normal interface (PERMISSIVE).

Reply:
Got it, thanks for the clarification @kyessenov 

+1 to John's point - PERMISSIVE in ztunnel is implemented very differently; no sniffing is involved. It's 100% based on the presence of a client cert or not (the TLS library will take care of verification).

Reply:
Will add DISABLE functionality in a follow up PR (to keep the diff manageable)

Reply:
I am not sure I follow how you can detect a client cert without sniffing. Some protocols don't send anything on the wire. But I don't have the full context.

Reply:
I probably have my wires crossed on how Envoy's sniffing works then

Reply:
HBONE is a different port. If its 15008 its TLS, if its anything else its plaintext. No sniffing